addons/l10n_it_edi/wizard/account_move_send.py

```diff
@@ -1,12 +1,216 @@
 # -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from base64 import b64encode
+from markupsafe import Markup
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.addons.l10n_it_edi.tools.xml_utils import format_errors
+from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import AccountEdiProxyError
 
 
 class AccountMoveSend(models.Model):
     _inherit = 'account.move.send'
 
-    def _get_mail_attachment_from_doc(self, doc):
-        if doc.edi_format_id.code == 'fattura_pa':
+    enable_l10n_it_edi = fields.Boolean(compute='_compute_enable_l10n_it_edi')
+    l10n_it_edi_checkbox = fields.Boolean(
+        'Tax Agency (Italy)',
+        compute='_compute_l10n_it_edi_checkbox',
+        store=True,
+        readonly=False,
+        help=(
+            "Send the invoice to the Italian Tax Agency.\n"
+            "It is set as readonly if a report has already been created, to avoid inconsistencies.\n"
+            "To re-enable it, delete the PDF attachment."
+        )
+    )
+    l10n_it_edi_readonly = fields.Boolean(compute='_compute_l10n_it_edi_readonly')
+    l10n_it_edi_warning = fields.Html(compute='_compute_l10n_it_edi_warning')
+
+    # -------------------------------------------------------------------------
+    # COMPUTE/CONSTRAINS METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('move_ids', 'enable_l10n_it_edi')
+    def _compute_send_mail_extra_fields(self):
+        # EXTENDS account
+        super()._compute_send_mail_extra_fields()
+        for wizard in self:
+            if (wizard.enable_send_mail
+                and wizard.enable_l10n_it_edi
+                and wizard.move_ids.filtered(lambda x: x._l10n_it_edi_is_self_invoice())):
+                wizard.enable_send_mail = False
+                wizard.checkbox_send_mail = False
+                wizard.send_mail_readonly = False
+                wizard.send_mail_warning_message = None
```

Comment:
- You must only set values for field specifically marked as computed by `_compute_send_mail_extra_fields`:

`enable_l10n_it_edi = fields.Boolean(compute='_compute_send_mail_extra_fields')`

- You should not depend of `enable_send_mail`.
Instead, you should depend of the PDF to know if you want to send something or look at your file to know if you need to resend something

Reply:
Fact is, I'm enabling the Send and Print button also for Vendor Bills (I need it), so I wanted to exclude those from having "Mail" or "Post" because it doesn't make any sense to send a Vendor Bill back to your vendor.

Reply:
> You should not depend of enable_send_mail.

I don't have any dependency on `enable_send_mail` for sending my stuff... ?
Why should I put `enable_l10n_it_edi` inside `_compute_send_mail_extra_fields` ?

I think I don't understand well, probably I'll ask you to have a call tomorrow.